src/libs/hooks/useOutsideClick.js

```diff
@@ -1,8 +1,10 @@
-import { useEffect, useState, useCallback } from 'react';
-
-export default function useOutsideClick(ref, handler) {
-    const [isListening, setIsListening] = useState(false);
+import { useEffect, useCallback } from 'react';
 
+/**
+ * This component implements the positioning of a component (inserted in the DOM at the body level) based on a trigger DOM element. By the way of example, you can think of the use case of a Menu Options, Tooltip, Popup that should be floating on top of all the elements and it should be correctly positioned based on the component/element that triggers the show/open action.
+ * @category Internal
+ */
+export default function useOutsideClick(ref, handler, isListening = true) {
     const listener = useCallback(
```

Comment:
it should depend on the handler, because it will be use like
```js
useOutsideClick(ref, () => {
}, isOpen);
```
then the handlers will be different every time the function executes but in reality, it will be the same

Reply:
same in use effect

Reply:
```js
event => {
    if (ref && isDomElement(ref.current) && !ref.current.contains(event.target)) {
        return handler(event);
    }
}
```